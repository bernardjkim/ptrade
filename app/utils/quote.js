/**
 * Format the given quote
 * TODO: Should this be a util funciton? Maybe just a separate component?
 *
 * @param  {object} quote An object containing the stock quote
 *
 * @return {array}        The formatted quote
 */
export function formatQuote(quote) {
  // currently only displaying these rows
  if (!quote) return [];
  return [
    { name: 'OPEN', value: `$${quote.open}` },
    { name: 'CLOSE', value: `$${quote.close}` },
    { name: 'HIGH', value: `$${quote.high}` },
    { name: 'LOW', value: `$${quote.low}` },
    { name: '52 WK HIGH', value: `$${quote.week52High}` },
    { name: '52 WK LOW', value: `$${quote.week52Low}` },
    {
      name: 'CHANGE',
      value: quote.change >= 0 ? `$${quote.change}` : `-$${-1 * quote.change}`,
    },
    {
      name: 'CHANGE %',
      value:
        quote.changePercent >= 0
          ? `%${quote.changePercent}`
          : `-%${-1 * quote.changePercent}`,
    },
    {
      name: 'AVG VOL',
      value: `${(quote.avgTotalVolume / 1000000).toPrecision(4)} M`,
    },
    {
      name: 'MKT CAP',
      value: `${Math.floor(quote.marketCap / 1000000)} M`,
    },
  ];
  // return [
  //   { name: 'OPEN', value: quote.open ? `$${quote.open}` : '' },
  //   { name: 'CLOSE', value: quote.close ? `$${quote.close}` : '' },
  //   { name: 'HIGH', value: quote.high ? `$${quote.high}` : '' },
  //   { name: 'LOW', value: quote.low ? `$${quote.low}` : '' },
  //   {
  //     name: '52 WK HIGH',
  //     value: quote.week52High ? `$${quote.week52High}` : '',
  //   },
  //   { name: '52 WK LOW', value: quote.week52Low ? `$${quote.week52Low}` : '' },
  //   {
  //     name: 'AVG VOL',
  //     value: quote.avgTotalVolume ? quote.avgTotalVolume : '',
  //   },
  //   { name: 'MKT CAP', value: quote.marketCap ? quote.marketCap : '' },
  // ];
}
