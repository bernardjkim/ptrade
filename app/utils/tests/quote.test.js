/**
 * Test the formatQuote function
 */
import { formatQuote } from '../quote';

describe('formatQuote', () => {
  it('should format and return the quote correctly', () => {
    const quote = {
      open: 20.59,
      close: 20.23,
      high: 21.06,
      low: 19.47,
      latestPrice: 20.23,
      change: 0.01,
      changePercent: 0.00049,
      avgTotalVolume: 116184509,
      marketCap: 20218007980,
      week52High: 34.14,
      week52Low: 9.04,
    };
    expect(formatQuote(quote)).toMatchSnapshot();
  });
});
