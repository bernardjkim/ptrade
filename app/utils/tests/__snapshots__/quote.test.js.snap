// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatQuote should format and return the quote correctly 1`] = `
Array [
  Object {
    "name": "OPEN",
    "value": "$20.59",
  },
  Object {
    "name": "CLOSE",
    "value": "$20.23",
  },
  Object {
    "name": "HIGH",
    "value": "$21.06",
  },
  Object {
    "name": "LOW",
    "value": "$19.47",
  },
  Object {
    "name": "52 WK HIGH",
    "value": "$34.14",
  },
  Object {
    "name": "52 WK LOW",
    "value": "$9.04",
  },
  Object {
    "name": "CHANGE",
    "value": "$0.01",
  },
  Object {
    "name": "CHANGE %",
    "value": "%0.00049",
  },
  Object {
    "name": "AVG VOL",
    "value": "116.2 M",
  },
  Object {
    "name": "MKT CAP",
    "value": "20218 M",
  },
]
`;
